client/dashboard/sites/site/index.tsx

```diff
@@ -48,7 +49,16 @@ function Site() {
 							items={ sites }
 							value={ site }
 							getItemName={ getSiteDisplayName }
-							getItemUrl={ ( site ) => `/sites/${ site.slug }` }
+							getItemUrl={ ( site ) => {
+								const currentPath = location.pathname;
+								const sitePattern = `/sites/${ siteSlug }`;
+
+								if ( currentPath.includes( sitePattern ) ) {
```

Comment:
I guess we can get the link from the last match 🤔

```js
const router = useRouter();
const matches = useMatches();
const lastMatch = matches[ matches.length - 1 ];

...

getItemUrl={ ( site ) =>
	router.buildLocation( {
		to: lastMatch.fullPath,
		params: { ...lastMatch.params, siteSlug: site.slug },
	} ).href
}
```

Also, we need to update the environment switcher as well 😄

Reply:
Proposedhttps://github.com/Automattic/wp-calypso/pull/106233